src/compas_viewer/scene/robotobject.py

```diff
@@ -0,0 +1,171 @@
+                if value:
+                    self.viewer.tree.add_object(collision_object, self.visual_objects[i - 1] if i > 0 else self)
+                    self.viewer.instance_colors[collision_object.instance_color.rgb255] = collision_object
+                else:
+                    self.viewer.tree.remove_object(collision_object)
+
+        def init(self):
+            """Initialize the robot object with creating the visual and collision objects."""
+            for i, visual_object in enumerate(self.visual_objects):
+                visual_object.init()
+                if self.show_visual:
+                    self.viewer.tree.add_object(visual_object, self.visual_objects[i - 1] if i > 0 else self)
+                    self.viewer.instance_colors[visual_object.instance_color.rgb255] = visual_object
+
+            for i, collision_object in enumerate(self.collision_objects):
+                collision_object.init()
+                if self.show_collision:
+                    self.viewer.tree.add_object(collision_object, self.visual_objects[i - 1] if i > 0 else self)
+                    self.viewer.instance_colors[collision_object.instance_color.rgb255] = collision_object
+
+        def transform(self, geometry, transformation: Transformation):
+            """Transform the geometry by a given transformation.
+
+            See Also
+            --------
+            :class:`compas_robots.scene.AbstractRobotModelObject`
+            """
+            geometry.transformation = transformation * geometry.transformation
+
+        def create_geometry(
+            self, geometry: Mesh, name: Optional[str] = None, color: Optional[Color] = None
+        ) -> MeshObject:
+            """Create a mesh object from a given geometry.
+
+            See Also
+            --------
+            :class:`compas_robots.scene.AbstractRobotModelObject`
+            """
+            mesh_object = MeshObject(  # type: ignore
+                geometry,
+                viewer=self.viewer,
+                parent=None,
+                is_selected=self.is_selected,
+                is_locked=self.is_locked,
+                is_visible=self.is_visible,
+                show_points=self.show_points,
+                show_lines=self.show_lines,
+                show_faces=self.show_faces,
+                pointscolor=self.pointscolor,
+                linescolor=self.linescolor,
+                facescolor=color or self.facescolor,
+                lineswidth=self.lineswidth,
+                pointssize=self.pointssize,
+                opacity=self.opacity,
+                config=self.viewer.scene_config,
+                hide_coplanaredges=self.hide_coplanaredges,
+                use_vertexcolors=self.use_vertexcolors,
+                name=name,
+                transformation=Transformation(),
+            )
+
+            return mesh_object
+
+        def update(self, joint_state: Optional[Configuration] = None, update_buffers: bool = False):
```

Comment:
I feel a bit strange here to allow to supply a joint_state into the update function. This makes it misaligned from the `update` of other objects. I would create a separated function and call it `update_joints`